reward/stakingManager.go

```diff
@@ -0,0 +1,104 @@
+package reward
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/event"
+	"github.com/klaytn/klaytn/params"
+)
+
+const (
+	chainHeadChanSize = 10
+)
+
+type stakingManager struct {
+	abm              *addressBookManager
+	stakingInfoCache *stakingInfoCache
+	governanceHelper governanceHelper
+	bc               *blockchain.BlockChain
+	chainHeadCh      chan blockchain.ChainHeadEvent
+	chainHeadSub     event.Subscription
+}
+
+func newStakingManager(bc *blockchain.BlockChain, governanceHelper governanceHelper) *stakingManager {
+	abm := newAddressBookManager(bc, governanceHelper)
+	sc := newStakingInfoCache()
+
+	return &stakingManager{
+		abm:              abm,
+		stakingInfoCache: sc,
+		governanceHelper: governanceHelper,
+		bc:               bc,
+		chainHeadCh:      make(chan blockchain.ChainHeadEvent, chainHeadChanSize),
+	}
+}
+
+// GetStakingInfoFromStakingCache returns corresponding staking information for a block of blockNum.
+func (sm *stakingManager) getStakingInfoFromStakingCache(blockNum uint64) *StakingInfo {
+	number := params.CalcStakingBlockNumber(blockNum)
+
+	stakingInfo, err := sm.updateStakingCache(number)
+	if err != nil {
+		logger.Error("Failed to get staking information", "Block number", blockNum, "number of staking block", number, "err", err)
+		return nil
+	}
+
+	if stakingInfo.BlockNum != number {
+		logger.Error("Invalid staking info from staking cache", "Block number", blockNum, "expected staking block number", number, "actual staking block number", stakingInfo.BlockNum)
+		return nil
+	}
+
+	logger.Debug("Staking cache hit.", "Block number", blockNum, "number of staking block", number)
+	return stakingInfo
+}
+
+// updateStakingCache updates staking cache with staking information of given block number.
+func (sm *stakingManager) updateStakingCache(blockNum uint64) (*StakingInfo, error) {
+	if cachedStakingInfo := sm.stakingInfoCache.get(blockNum); cachedStakingInfo != nil {
```

Comment:
How about move below code into `getStakingInfoFromStakingCache`?
Because, this function name is `updateStakingCache`. :)
```
	if cachedStakingInfo := sm.stakingInfoCache.get(blockNum); cachedStakingInfo != nil {
		// already updated
		return cachedStakingInfo, nil
	}
```

Reply:
reflected